protostar/starknet_gateway/gateway_facade.py

```diff
@@ -164,6 +164,15 @@ async def deploy_via_udc(
         except TransactionFailedError as ex:
             raise TransactionException(str(ex)) from ex
         except ClientError as ex:
+            account_address_hex = hex(int(account_address))
```

Comment:
You can use `account_address` directly in a string without creating hex representation. `account_address` is an `Account` value object, which has overwritten `__str__` method.

Reply:
The problem is that this `__str__` method prints the hex with all the unnecessary zeros so instead of `0xf50bc325` you have `0x00000000000000000000000000000000000000000000000000000000f50bc325`. It's hard to do pattern matching in such a case.

Anyway, `account_address` was passed as an int although the typing says it's `Address`. Weird that type checker didn't see that.

Reply:
Please let me know if this can be resolved.